Assets/BowAndArrowImproved/Scripts/Traps/TrapActivation.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using Unity.XR.CoreUtils;
+using UnityEngine;
+
+public class TrapActivation : MonoBehaviour
+{
+    [SerializeField] private float activationTime = 3f;
+    [SerializeField] private List<ContinuousTrap> myTraps;
+    [SerializeField] private LayerMask hitMask;
+    [SerializeField] private bool areTrapsActive;
+
+    private void OnTriggerEnter(Collider other)
+    {
+        if (!hitMask.Contains(other.attachedRigidbody.gameObject.layer) || areTrapsActive) return;
+
+        areTrapsActive = true;
+        foreach (ContinuousTrap trap in myTraps)
+        {
+            trap.ToggleTrapActivation(true);
+        }
+
+        StartCoroutine(CountdownAndDeactivateTrap());
```

Comment:
keep a reference to that coroutine and cancel it in onDestroy in case this object is destroyed before the coroutine ends to prevent memory leaks and other problems...

Reply:
https://docs.unity3d.com/2020.3/Documentation/Manual/Coroutines.html
no need, according to unity all coroutine will stop when Destroy is called.